gitautosync/pull_from_remote.py

```diff
@@ -0,0 +1,129 @@
+import os
+import re
+import subprocess
+from .util import logger
+
+DELETED_FILE_REGEX = re.compile(
+    r"deleted:\s+"  # Look for deleted: + any amount of whitespace...
+    r"([^\n\r]+)"   # and match the filename afterward.
+)
+
+
+def pull_from_remote(git_url, branch_name, repo_dir):
```

Comment:
Can you make a class which has as its instance data the git url, branch name and repo dir, and make these all methods of the class? that's cleaner than passing that state around like this IMO.